Remove optional from dcc feign and default local value for valuesets

alexmoroianu opened the pull request.

alexmoroianu committed:
Remove optional from dcc feign and default local value for valuesets

roesslerj commented on services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/DgcStructureProviderDefaultValuesetsMissingTest.java:
```diff
@@ -44,7 +46,8 @@
 
   @Test
-  void default_value_missing_should_result_in_empty_dir() throws UnableToLoadFileException, FetchValueSetsException {
+  @Disabled("Default valuesets are deprecated.")
```
I would delete the test instead.

roesslerj commented on services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/DgcStructureProviderDefaultValuesetsMissingTest.java:
```diff
@@ -20,6 +21,7 @@
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
+@Deprecated
```
At what time would you delete the class? I am in favor or stating that explicitly, otherwise nobody will be bold enough to actually do it.

roesslerj commented on services/distribution/src/test/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateJsonToProtobufTest.java:
```diff
@@ -26,6 +26,7 @@
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
+@Deprecated
```
Same as above... state when it's safe to delete.

roesslerj commented on services/distribution/src/test/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateJsonToProtobufTest.java:
```diff
@@ -51,7 +52,8 @@ void setup(){
 
   @Test
-  void shouldReadDefaultMahJsonIfNotConfigured() throws UnableToLoadFileException, FetchValueSetsException {
+  @Disabled("Default valuesets are deprecated.")
```
Test will probably never be used again, so delete.

hilmarf committed:
Merge branch 'main' into refactor/valuesets_not_fetched

alexmoroianu committed:
REMOVE - deprecated tests

alexmoroianu committed:
Merge branch 'refactor/valuesets_not_fetched' of github.com-alexmoroianu:corona-warn-app/cwa-server into refactor/valuesets_not_fetched

FelixRottler commented on services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:
```diff
@@ -63,20 +63,14 @@ public DigitalGreenCertificateStructureProvider(DistributionServiceConfig distri
-    for (String currentLanguage: dgcConfig.getSupportedLanguages()) {
-      try {
+    try {
+      for (String currentLanguage : dgcConfig.getSupportedLanguages()) {
```
Not sure but isnt this exactly what an indexingDirectory is used for? Providing indexing (in this case languages) and writing the same logic for each index and add its result to the directory? Maybe we can have a call about that

FelixRottler commented on services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:
```diff
@@ -63,20 +63,14 @@ public DigitalGreenCertificateStructureProvider(DistributionServiceConfig distri
 
-  private DirectoryOnDisk constructArchiveToPublish(DigitalGreenCertificate dgcConfig)
-      throws UnableToLoadFileException {
+  private DirectoryOnDisk constructArchiveToPublish(DigitalGreenCertificate dgcConfig) {
```
Maybe we can find a better naming for the methodname and the parameter. Or is DigitalGreenCertificate a config file?

alexmoroianu replied:
DigitalGreenCertificate is a config. For me the method name seems okay. Do you have any suggestion ?

FelixRottler replied:
no it is alright for me :)

FelixRottler commented on services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java:
```diff
@@ -44,42 +43,32 @@ public DigitalGreenCertificateToCborMapping(DigitalCovidCertificateClient digita
-   *                                          be fetched, json validation fails for a specific rule etc. This exception
-   *                                          will propagate and will stop any archive to be published down in the
-   *                                          execution.
+   * @throws DigitalCovidCertificateException - thrown if json validation schema is not found or the validationfails for
```
space missing (validationfails)

alexmoroianu replied:
I don't really figure out what to change here. The validation on the PR seems to pass.

FelixRottler replied:
just a typo :)  thrown if json validation schema is not found or the validationfails for
```suggestion
   * @throws DigitalCovidCertificateException - thrown if json validation schema is not found or the validation fails for
```

FelixRottler commented on services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java:
```diff
@@ -44,42 +43,32 @@ public DigitalGreenCertificateToCborMapping(DigitalCovidCertificateClient digita
     List<BusinessRuleItem> businessRulesItems = digitalCovidCertificateClient.getRules();
     List<BusinessRule> businessRules = new ArrayList<>();
 
     for (BusinessRuleItem businessRuleItem : businessRulesItems) {
```
Same thing here with the indexingdirectory

roesslerj committed:
Renaming DigitalGreenCertificateStructureProvider to DigitalCertificatesStructureProvider

as it contains both the DigitalGreenCertificates logic, as well as the DigitalSigningCertificates

alexmoroianu committed:
Merge branch 'main' into refactor/valuesets_not_fetched

roesslerj committed:
For some reason, this was left missing...

alexmoroianu committed:
Merge branch 'feature/update-classnames' into refactor/valuesets_not_fetched

alexmoroianu committed:
FIX - structure provider renaming

ioangut committed:
Merge branch 'main' into refactor/valuesets_not_fetched

FelixRottler committed:
Update services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java

alexmoroianu committed:
FIX - validations
